modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java

```diff
@@ -0,0 +1,281 @@
+/*
+*Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*WSO2 Inc. licenses this file to you under the Apache License,
+*Version 2.0 (the "License"); you may not use this file except
+*in compliance with the License.
+*You may obtain a copy of the License at
+*
+*http://www.apache.org/licenses/LICENSE-2.0
+*
+*Unless required by applicable law or agreed to in writing,
+*software distributed under the License is distributed on an
+*"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*KIND, either express or implied.  See the License for the
+*specific language governing permissions and limitations
+*under the License.
+*/
+package org.wso2.carbon.registry.es.publisher.lifecycle;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.wink.client.ClientResponse;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+import org.wso2.carbon.governance.lcm.stub.LifeCycleManagementServiceExceptionException;
+import org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.registry.es.utils.GregESTestBaseTest;
+import org.wso2.carbon.registry.es.utils.LifeCycleBean;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceExceptionException;
+import org.wso2.greg.integration.common.clients.LifeCycleAdminServiceClient;
+import org.wso2.greg.integration.common.clients.LifeCycleManagementClient;
+import org.wso2.greg.integration.common.clients.ManageGenericArtifactAdminServiceClient;
+import org.wso2.greg.integration.common.clients.ResourceAdminServiceClient;
+import org.wso2.greg.integration.common.utils.GenericRestClient;
+import org.wso2.carbon.automation.test.utils.common.FileManager;
+import org.wso2.carbon.registry.es.utils.LifeCycleConstants;
+import org.wso2.carbon.registry.es.utils.LifeCycleUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+/***
+ * This class will test Greg Publisher Lifecycle State Duration and Its Color bellow the asset icon
+ */
+public class GRegPublisherLCStateDurationColorBoxTestCase extends GregESTestBaseTest {
+    public static final String CLASS_NAME = GRegPublisherLCStateDurationTestCase.class.getName();
+    private static final Log log = LogFactory.getLog(GRegPublisherLCStateDurationTestCase.class);
+    String assetId;
```

Comment:
Thank you Nuwan for mentioning this,
This entire code block has been refactored with new time in lifecycle feature.